### PR TITLE
HARP-6251: Clip the polygons.

### PR DESCRIPTION
**Pull request by robertoraggi**
Clip the polygon geometries when using fill and standard techniques.

Signed-off-by: Roberto Raggi <roberto.raggi@here.com>

### CODE REVIEW

**Comment by robertoraggi** [comment]
@heremaps-bot retest this please

**Comment by robertoraggi** [comment]
@heremaps-bot retest this please

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 1]
```diff
@@ -631,13 +632,33 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
...
+                    if (!isExtrudedPolygonTechnique(technique)) {
+                        if (!clipRing) {
+                            clipRing = [
+                                new THREE.Vector2(0, 0),
+                                new THREE.Vector2(extents, 0),
+                                new THREE.Vector2(extents, extents),
+                                new THREE.Vector2(0, extents)
+                            ];
+                        }
+                        coords = clipPolygon(coords, clipRing);
```
Comment (ninok) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: Maybe I missed it but I cannot see any code handling the "pseudo" edges introduced by clipping. Will it work if we render the polygon with outline or will we see an outline on the tile border?
An image regression test for that case would also be highly appreciated.

Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: The way stroking is implemented is just not working reliably when tiles are clipped and we have sources where tiles are clipped. If we want to support stroking polygons we may need to rewrite that code, maybe using a similar approach as the one we use for outlines.

**Pull request merged**